app/services/import_service.rb

```diff
@@ -3,24 +3,27 @@
 class ImportService
   include Diaspora::Logging
 
-  def import_by_user(user, opts={})
-    import_by_files(user.export.current_path, user.exported_photos_file.current_path, user.username, opts)
+  def import_by_user(user_name, import_parameters)
+    profile_path = import_parameters["profile_path"]
+    photos_path = import_parameters["photos_path"]
+
+    import_by_files(profile_path, photos_path, user_name)
   end
 
   def import_by_files(path_to_profile, path_to_photos, username, opts={})
+    user = User.find_by(username: username)
+    raise ArgumentError, "Username #{username} should exist before uploading photos." if user.nil?
```

Comment:
I think raising exception from the sidekiq job would schedule this job to retry. If this error is irrecoverable we should instead finish the job without errors, possibly logging some error messages.